processor.go

```diff
@@ -215,6 +215,23 @@ func (proc *Processor) Encode(text string) []Token {
 		return -1
 	})
 
+	buf := make([]byte, 64) // reusable buffer for merging symbols
+	mergeSymbols := func(x, y symListElem) (string, int, bool) {
+		sz := len(x.symbol) + len(y.symbol)
+		if cap(buf) < sz {
+			buf = append(buf[:0], make([]byte, sz)...)
```

Comment:
This seems a bit obscure without a comment.
I don't think tokens larger than 64 bytes are very likely anyway (and the 64 number can probably be increased without harm?), so would it really hurt performance to have a more straightforward reallocation here? (assign larger buf)

Reply:
Will do. Old habits die hard. :)

One could precalculate the longest string piece when loading and store that length in the processor, and simplify all this even more.

What do you think?

Reply:
Yes, that sounds good!

Reply:
Done